app/frontend/wallet/blockchain-explorer.ts

```diff
@@ -170,6 +179,114 @@ const blockchainExplorer = (ADALITE_CONFIG) => {
     })
   }
 
+  async function getPoolInfo(url) {
+    const response = await request(
+      `${ADALITE_CONFIG.ADALITE_SERVER_URL}/api/poolMeta`,
+      'POST',
+      JSON.stringify({poolUrl: url}),
+      {'Content-Type': 'application/json'}
+    ).catch(() => {
+      return {}
+    })
+    return response
+  }
+
+  async function getDelegationHistory(accountPubkeyHex, validStakepools) {
+    // Urls
+    const delegationsUrl = `https://explorer-staging.adalite.io/api/account/delegationHistory/${accountPubkeyHex}`
+    // const rewardsUrl = `https://explorer-staging.adalite.io/api/account/rewardHistory/${accountPubkeyHex}`
+    const withdrawalsUrl = `https://explorer-staging.adalite.io/api/account/withdrawalHistory/${accountPubkeyHex}`
+
+    // Run requests in parallel
+    const delegationsRequest = request(delegationsUrl).catch(() => [])
+    // const rewardsRequest = request(rewardsUrl).catch(()=>[]
+    const withdrawalsRequest = request(withdrawalsUrl).catch(() => [])
+
+    // Await all
+    const delegations = await delegationsRequest
+    const rewards = [] //await rewardsRequest
+    const withdrawals = await withdrawalsRequest
```

Comment:
is there a reason the requests are put into variables and are not ran inline?

Reply:
Refactored